SharpTox/HL/Transfers/ToxTransfer.cs

```diff
@@ -12,15 +12,39 @@ public class ToxTransfer
         public string Name { get; private set; }
         public long Size { get; private set; }
         public ToxFileKind Kind { get; private set; }
-        public ToxTransferState State { get; protected set; } //initial state is 'paused' for the receiving end, 'in progress' for the sending end
+        public ToxTransferState State { get; protected set; }
+
+        private int _transferredBytes;
+
+        protected int TransferredBytes
+        {
+            get { return _transferredBytes; }
+            set
+            {
+                if (value == _transferredBytes)
+                    return;
+                _transferredBytes = value;
+                if (ProgressChanged != null)
+                    ProgressChanged(this, new ToxTransferEventArgs.ProgressEventArgs(Progress));
```

Comment:
This event is going to be fired **a lot**. Perhaps we should make it so that it only fires every time the percentage goes up by a full percent. So for 4% and 5%, but not for 4%, 4.2%, 4.4%...5% and so on.